### ISSUE TITLE
Why are fields (constants) disallowed in Annotation body declaration?

### ISSUE THREAD

**Issue by amihaiemil** [javaparser/javaparser#3791]
In Java, it is valid to have constant declarations inside annotation body:

```java
//this is valid according to the Java grammar and IntelliJ also doesn't complain
public @interface TestAnnotation {
    public static final int CONSTANT = 0;

}
```

However, JavaParser forbids this - added in [this](https://github.com/javaparser/javaparser/commit/1235b68df864ab6508870e20910047cc9d0a1213) commit, with [this](https://github.com/javaparser/javaparser/pull/2153) PR, for Issue https://github.com/javaparser/javaparser/issues/676.

I'm curious what is the reason behind this? I know it's a long shot, but maybe @ftomassetti you remember what was the reason for Issue 676?

**Comment by ftomassetti** [comment]
That is strange indeed. I guess I just did not know annotation well enough. I checked the JLS for Java 8 and constants were already available at that time. The issue was opened in 2016, but I think that already at that time annotations could have constants. 

So I think that was just a very common instance of me being wrong :D

**Comment by ftomassetti** [comment]
Relevant part of the JLS: https://docs.oracle.com/javase/specs/jls/se8/html/jls-9.html#jls-9.6

**Comment by amihaiemil** [comment]
@ftomassetti I could make a PR to revert this change. I'm not sure of your work process, though, how should I proceed? :D

**Comment by ftomassetti** [comment]
I am not involved in the project anymore, so that is a question for the maintainers :)
I suppose they would be happy to get a PR

**Comment by jlerbsc** [comment]
>

**Comment by jlerbsc** [comment]
> In Java, it is valid to have constant declarations inside annotation body:

This may be because there are restrictions on the members of the annotation which cannot be specified by the NodeWithMembers interface. For example a constant must be initialized.

**Comment by jlerbsc** [comment]
This issue is fixed in version 3.25.3.
